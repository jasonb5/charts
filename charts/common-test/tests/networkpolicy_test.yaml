suite: test networkpolicy
values:
  - values/common.yaml
templates:
  - app.yaml
chart:
  appVersion: 0.1.0
tests:
  - it: should disable defaults
    set:
      workload: deployment

      networkPolicy:
        enabled: true
        disableDefaultEgress: true
        disableDefaultIngress: true
        egress:
          - to:
              - namespaceSelector:
                  matchLabels:
                    kubernetes.io/metadata.name: production
            ports:
              - protocol: TCP
                port: 3306
        ingress:
          - from:
              - namespaceSelector:
                  matchLabels:
                    kubernetes.io/metadata.name: production
            ports:
              - protocol: TCP
                port: 8000
      service:
        ports:
          default:
            port: 80
    asserts:
      - equal:
          path: spec
          value:
            egress:
              - to:
                  - namespaceSelector:
                      matchLabels:
                        kubernetes.io/metadata.name: production
                ports:
                  - protocol: TCP
                    port: 3306
            ingress:
              - from:
                  - namespaceSelector:
                      matchLabels:
                        kubernetes.io/metadata.name: production
                ports:
                  - protocol: TCP
                    port: 8000
            podSelector:
              app.kubernetes.io/instance: RELEASE-NAME
              app.kubernetes.io/name: common-test
              common/workload: default
            policyTypes:
              - Egress
              - Ingress
        documentIndex: 2
      - hasDocuments:
          count: 3
  - it: should merge with defaults
    set:
      workload: deployment

      networkPolicy:
        enabled: true
        egress:
          - to:
              - namespaceSelector:
                  matchLabels:
                    kubernetes.io/metadata.name: production
            ports:
              - protocol: TCP
                port: 3306
        ingress:
          - from:
              - namespaceSelector:
                  matchLabels:
                    kubernetes.io/metadata.name: production
            ports:
              - protocol: TCP
                port: 8000
      service:
        ports:
          default:
            port: 80
    asserts:
      - equal:
          path: spec
          value:
            egress:
              - ports:
                  - port: 3306
                    protocol: TCP
                to:
                  - namespaceSelector:
                      matchLabels:
                        kubernetes.io/metadata.name: production
              - {}
            ingress:
              - from:
                  - namespaceSelector:
                      matchLabels:
                        kubernetes.io/metadata.name: production
                ports:
                  - port: 8000
                    protocol: TCP
              - from:
                  - podSelector: {}
                ports:
                  - port: 80
                    protocol: TCP
            podSelector:
              app.kubernetes.io/instance: RELEASE-NAME
              app.kubernetes.io/name: common-test
              common/workload: default
            policyTypes:
              - Egress
              - Ingress
        documentIndex: 2
      - hasDocuments:
          count: 3
  - it: should create default policy
    set:
      workload: deployment

      networkPolicy:
        enabled: true

      service:
        ports:
          default:
            port: 80
    asserts:
      - isKind:
          of: NetworkPolicy
        documentIndex: 2
      - isKind:
          of: Service
        documentIndex: 1
      - isKind:
          of: Deployment
        documentIndex: 0
      - equal:
          path: spec
          value:
            egress:
              - {}
            ingress:
              - from:
                  - podSelector: {}
                ports:
                  - port: 80
                    protocol: TCP
            podSelector:
              app.kubernetes.io/instance: RELEASE-NAME
              app.kubernetes.io/name: common-test
              common/workload: default
            policyTypes:
              - Egress
              - Ingress
        documentIndex: 2
      - hasDocuments:
          count: 3
