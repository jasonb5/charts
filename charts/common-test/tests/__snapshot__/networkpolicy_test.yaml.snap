should create default policy:
  1: |
    apiVersion: networking.k8s.io/v1
    kind: NetworkPolicy
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: common-test
        app.kubernetes.io/version: 0.1.0
        helm.sh/chart: common-test-0.1.0
      name: RELEASE-NAME-common-test
      namespace: NAMESPACE
    spec:
      egress:
        - {}
      ingress:
        - from:
            - podSelector: {}
          ports:
            - port: 80
              protocol: TCP
      podSelector:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/name: common-test
      policyTypes:
        - Egress
        - Ingress
should disable defaults:
  1: |
    apiVersion: networking.k8s.io/v1
    kind: NetworkPolicy
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: common-test
        app.kubernetes.io/version: 0.1.0
        helm.sh/chart: common-test-0.1.0
      name: RELEASE-NAME-common-test
      namespace: NAMESPACE
    spec:
      egress:
        - ports:
            - port: 3306
              protocol: TCP
          to:
            - namespaceSelector:
                matchLabels:
                  kubernetes.io/metadata.name: production
      ingress:
        - from:
            - namespaceSelector:
                matchLabels:
                  kubernetes.io/metadata.name: production
          ports:
            - port: 8000
              protocol: TCP
      podSelector:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/name: common-test
      policyTypes:
        - Egress
        - Ingress
should merge with defaults:
  1: |
    apiVersion: networking.k8s.io/v1
    kind: NetworkPolicy
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: common-test
        app.kubernetes.io/version: 0.1.0
        helm.sh/chart: common-test-0.1.0
      name: RELEASE-NAME-common-test
      namespace: NAMESPACE
    spec:
      egress:
        - ports:
            - port: 3306
              protocol: TCP
          to:
            - namespaceSelector:
                matchLabels:
                  kubernetes.io/metadata.name: production
        - {}
      ingress:
        - from:
            - namespaceSelector:
                matchLabels:
                  kubernetes.io/metadata.name: production
          ports:
            - port: 8000
              protocol: TCP
        - from:
            - podSelector: {}
          ports:
            - port: 80
              protocol: TCP
      podSelector:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/name: common-test
      policyTypes:
        - Egress
        - Ingress
